DOC: Drafted new documentation page outlining how to create inset maps and indicators

moss-xyz opened the pull request:
As previously discussed in #3487, I've drafted an entirely new page to add to the documentation that outlines how to create inset maps and indicators using both the base `matplotlib` package, as well as `matplotlib-map-utils`, which has some additional functionality in it.

Apologies if I've done this wrong - it seems it pulled all my previous commits from the last time I did documentation, so sorry if that complicates the git history. Happy to fix if you tell me how.

Otherwise, take a look at the new page and let me know if I need to change anything!

fivemservereyestore commented:
The documentation in this repository is excellent! Very clear and comprehensive.

myeyestore commented:
The documentation in this repository is excellent! Very clear and comprehensive.

m-richards commented on doc/source/gallery/inset_map.ipynb:
```diff
@@ -0,0 +1,442 @@
+    "nybb = nybb.to_crs(32619)  # Convert the dataset to a coordinate system which uses meters\n",
+    "\n",
+    "ax = nybb.plot()\n",
+    "# Demonstrating the type of the axis object\n",
```
```suggestion
```

m-richards commented on doc/source/gallery/inset_map.ipynb:
```diff
@@ -0,0 +1,442 @@
+    "#### Adding an Inset Axis\n",
+    "The inset axis (or \"child axis\", relative to the \"parent axis\" created by `.plot()`) can be added via the `inset_axes()` method on an `axes.Axes` object.\n",
+    "\n",
+    "This function's first argument, `bounds`, requires a 4-tuple of integers of floats, matching `[x0, y0, width, height]`, where `(x0, y0)` is the lower-left corner of the inset axes, and `width` and `height` are the width and height of the inset axes. By default, these values should be in `ax.transAxes`, i.e. ranging from zero to one, representing a fraction of the parent axis, though users ma ...
```
```suggestion
    "This function's first argument, `bounds`, requires a 4-tuple of integers of floats, matching `(x0, y0, width, height)`, where `(x0, y0)` is the lower-left corner of the inset axes, and `width` and `height` are the width and height of the inset axes. By default, these values should be in `ax.transAxes` scaling, i.e. ranging from zero to one, representing a fraction of the parent axis, though users may pass their own desired transformation via the `transform` argument.\n",
```

m-richards commented on doc/source/gallery/inset_map.ipynb:
```diff
@@ -0,0 +1,442 @@
+    "\n",
+    "This function's first argument, `bounds`, requires a 4-tuple of integers of floats, matching `[x0, y0, width, height]`, where `(x0, y0)` is the lower-left corner of the inset axes, and `width` and `height` are the width and height of the inset axes. By default, these values should be in `ax.transAxes`, i.e. ranging from zero to one, representing a fraction of the parent axis, though users ma ...
+    "\n",
+    "Many other arguments are possible, and `inset_axes()` will also accept any other kwarg as the normal Axes class. See [this documentation](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.inset_axes.html) for details.\n",
```
```suggestion
    "`inset_axes()` shares common kwargs with the regular `Axes` class along with others; see [the documentation](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.inset_axes.html) for details.\n",
```

m-richards commented on doc/source/gallery/inset_map.ipynb:
```diff
@@ -0,0 +1,442 @@
+    "\n",
+    "  - If integers or floats are passed to width and height, they are interpreted as *inches*. If strings are passed, they are interpreted as percentages of the parent axes, or to the bounding box specified under `bbox_to_anchor`.\n",
+    "\n",
+    "  - The `loc` keyword is similar to the `loc` keyword of `Legend` objects. It can be a string or an integer. See [this documentation](https://matplotlib.org/stable/api/legend_api.html#matplotlib.legend.Legend) for more details.\n",
+    "\n",
+    "  - Advanced positioning is also possible via `bbox_to_anchor`, and `bbox_transform`, just like `Legend`s. See [this documentation](https://matplotlib.org/stable/api/_as_gen/mpl_toolkits.axes_grid1.inset_locator.inset_axes.html) for more details.\n",
```
```suggestion
    "  - The `loc` keyword is similar to the `loc` keyword of `Legend` objects. It can be a string or an integer. See [the documentation](https://matplotlib.org/stable/api/legend_api.html#matplotlib.legend.Legend) for more details.\n",
    "\n",
    "  - Advanced positioning is also possible via `bbox_to_anchor`, and `bbox_transform`, just like `Legend`s. See [the documentation](https://matplotlib.org/stable/api/_as_gen/mpl_toolkits.axes_grid1.inset_locator.inset_axes.html) for more details.\n",
```

m-richards commented on doc/source/gallery/inset_map.ipynb:
```diff
@@ -0,0 +1,442 @@
+    "iax = mil.inset_axes(ax, width=1, height=1, loc=\"upper left\", borderpad=1, axes_kwargs={\"xticks\":[], \"yticks\":[]})\n",
+    "# Plotting a specific neighborhood on the inset axis\n",
+    "nyc = gpd.read_file(gda.data.geoda.nyc.url).to_crs(32619)\n",
+    "nyc.query(\"name=='Bushwick'\").plot(ax=iax)\n",
```
I suspect this is used in the inset for convenience, but I do think it somewhat distracts from the example by having the inset not match the zoomed out view. Could you do the inset with an island (or a few), to make this relationshipg clearer (perhaps just gdf.clip by a shapely box?).

moss-xyz replied:
Yep just updated this

m-richards commented on doc/source/gallery/inset_map.ipynb:
```diff
@@ -0,0 +1,442 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This library is intended to provide an easier-to-use interface for constructing inset maps (along with scale bars and north arrows) for matplotlib plots than either the `inset_axes()` method or function. It also provides \n",
```
"It also provides" the text cuts off here.

moss-xyz replied:
Fixed

m-richards commented on doc/source/gallery/inset_map.ipynb:
```diff
@@ -0,0 +1,442 @@
+   "outputs": [
+    {
+     "data": {
+      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAcsAAAGsCAYAAACy84ylAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjEsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvc2/+5QAAAAlwSFlzAAAPYQAAD2EBqD+naQAAlW9JREFUeJzs3XdYk+f6B/Dvm00CYW/ZCAguxL3qbmv3Hna35/TXaYcd9rTndJ3aYXvsbrVTbbXDaltt1WrdExEEQZC990hCQvb7+wONRAIkkJAE7s91cV0mvOMBgft91n0zLMuyIIQQQkiPOM5uACGEEOLqKFgSQgghfaBgSQghhPSBgiUhhBDSBwqWhBBCSB8oWBJCCCF9oGBJCCG ...
```
I would suggest this and the following example use keyword arguments here, to make clear which CRS goes with which - obviously they're both the same here, but I don't like seeing magic numbers going into functions by argument order.

moss-xyz replied:
Understood, fixed!

m-richards commented on doc/source/gallery/inset_map.ipynb:
```diff
@@ -0,0 +1,442 @@
+   ],
+   "source": [
+    "nybb = gpd.read_file(gda.get_path(\"nybb\"))\n",
+    "nybb = nybb.to_crs(32619)  # Convert the dataset to a coordinate system which uses meters\n",
```
this is super minor, but can we use `to_crs(epsg=32619)` or `to_crs("EPSG:32619")` for consistency with the rest of the documentation? Though now I notice this is consistent with the matplotlib scalebar page, but I think it would be better to standardise with epsg explicit in either form.

moss-xyz replied:
Not minor at all - fixed, let me know if I need to do the same for the scalebar page (I believe I was following the prior example with the code)

m-richards reviewed (commented):
@moss-xyz I've had a bit of a look through with some initial comments.

RE the extra commits, these are here for two reasons:
1. You're doing all your commits on main on your fork - this means that the changes from the scalebar PR are already in main on your fork and they're added on top here, as additional commits
2. These show up as additional commits because even though the scalebar stuff was merged, it was squash commited to clean up the git history - but that means that git can't see that the associated commits on your fork are already included and should be excluded here.

My suggestion here would be that when working on your fork, you create branches over there too for each distinct merge request you work on (but no need to worry about that for this one). This will make things cleaner, and if you ever have two or more merge requests going at once you won't pollute the merge request submitted with changes from both.

moss-xyz commented:
@m-richards thanks for taking the time to review, I *believe* I have addressed all of your comments now!

And thank you for the tip re: github, I'll be more intentional on any future edits I make :)

m-richards reviewed (approved):
Thanks @moss-xyz looking good to me now!

martinfleis commented:
The links wrapping code do not render properly on RTD - see https://geopandas--3557.org.readthedocs.build/en/3557/gallery/inset_map.html Can you omit code formatting when adding links?
